Description: Fix table stops
 ssldump segfaults when decoding some unknown enumerated values
 because decoder tables end with 0 instead of -1 expected by
 table search routines
Author: Zdenek Salvet <salvet@ics.muni.cz>
Bug-Debian: http://bugs.debian.org/627884
Last-Update: 2016-03-11
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- ssl/ssl.enums.c
+++ ssl/ssl.enums.c
@@ -151,7 +151,7 @@ decoder ContentType_decoder[]={
 		"application_data",
 		decode_ContentType_application_data
 	},
-{0}
+{-1}
 };
 
 static int decode_HandshakeType_HelloRequest(ssl,dir,seg,data)
@@ -498,7 +498,7 @@ decoder HandshakeType_decoder[]={
 		"Finished",
 		decode_HandshakeType_Finished
 	},
-{0}
+{-1}
 };
 
 decoder cipher_suite_decoder[]={
@@ -728,7 +728,7 @@ decoder AlertLevel_decoder[]={
 		"fatal",
 		decode_AlertLevel_fatal
 	},
-{0}
+{-1}
 };
 
 static int decode_AlertDescription_close_notify(ssl,dir,seg,data)
@@ -1031,7 +1031,7 @@ decoder AlertDescription_decoder[]={
 		"no_renegotiation",
 		decode_AlertDescription_no_renegotiation
 	},
-{0}
+{-1}
 };
 
 decoder compression_method_decoder[]={
@@ -1095,6 +1095,6 @@ decoder client_certificate_type_decoder[
 		"dss_fixed_dh",
 		decode_client_certificate_type_dss_fixed_dh
 	},
-{0}
+{-1}
 };
 
--- ssl/ssl_enum.c
+++ ssl/ssl_enum.c
@@ -70,7 +70,7 @@ decoder ContentType_decoder[]={
 		"application_data",
 		decode_ContentType_application_data
 	},
-{0}
+{-1}
 };
 
 static int decode_HandshakeType_hello_request(ssl,dir,seg,data)
@@ -260,7 +260,7 @@ decoder HandshakeType_decoder[]={
 		"finished",
 		decode_HandshakeType_finished
 	},
-{0}
+{-1}
 };
 
 decoder cipher_suite_decoder[]={
